packages/automated_actions/automated_actions/celery/external_resource/tasks.py

```diff
@@ -30,13 +36,73 @@ def external_resource_rds_reboot(
     force_failover: bool,
     action: Action,  # noqa: ARG001
 ) -> None:
-    try:
-        rds = get_external_resource(account=account, identifier=identifier)
-    except ExternalResourceAppInterfaceError as e:
-        raise HTTPException(status_code=404, detail=str(e)) from None
+    rds = get_external_resource(
+        account=account,
+        identifier=identifier,
+        provider=ExternalResourceProvider.RDS,
+    )
 
     credentials = get_aws_credentials(
         vault_secret=rds.account.automation_token, region=rds.account.region
     )
     with AWSApi(credentials=credentials, region=rds.region) as aws_api:
         ExternalResourceRDSReboot(aws_api, rds).run(force_failover=force_failover)
+
+
+class ExternalResourceFlushElastiCache:
+    def __init__(self, oc: OpenshiftClient, elasticache: ExternalResource) -> None:
+        self.oc = oc
+        self.elasticache = elasticache
+
+    def run(
+        self,
+        image: str,
+        command: list[str],
+        args: list[str],
+        secret_name: str,
+        env_secret_mappings: dict[str, str],
+    ) -> None:
+        job = job_builder(
+            image=image,
+            command=command,
+            args=args,
+            job_name_prefix="flush-elasticache-",
+            env_secrets={
+                key: SecretKeyRef(
+                    secret=secret_name,
+                    key=value,
+                )
+                for key, value in env_secret_mappings.items()
+            },
+        )
+        return self.oc.run_job(namespace=self.elasticache.namespace, job=job)
```

Comment:
how do we handle failed job and cleanup jobs?

Reply:
The automatic cleanup is configured via [ttl_seconds_after_finished](https://github.com/chassing/automated-actions/blob/4ed2123174670a8fae35e7aa5722cc1f20d907f3/packages/automated_actions_utils/automated_actions_utils/openshift_client.py#L107) job spec. This [handles](https://kubernetes.io/docs/concepts/workloads/controllers/ttlafterfinished/#cleanup-for-finished-jobs) it.